### ISSUE TITLE
Update README for new docs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6953]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#6953|
|Type|enhancement|
|Created|2014-02-09 10:39:07Z|
|Branch|https://github.com/twisted/twisted/tree/update-readme-6953-2|

The README still talks about documentation from /doc/ but now all documentatio is in /docs/

The trunk still have some leftover files in /doc/ ... they should be removed.

I have also a small diff for docs/_extensions/apilinks.py ... maybe it can be included together with this ticket... otherwise I can create a new ticket for that.

https://gist.github.com/adiroiban/8897350

Thanks!

<details><summary>Searchable metadata</summary>

```
trac-id__6953 6953
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__ 
branch__branches_update_readme_6953_2 branches/update-readme-6953-2
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1391942347000000 1391942347000000
changetime__1410787733165425 1410787733165425
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

Maybe also include a note about how to build the documentation.... 
```
cd docs
make html
```

There are many make targets and someone might be confused about which one to use.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

Ah...

Maybe also add docs/_build to .gitignore

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

(In [41592]) Branching to update-readme-6953.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

I'll take this one.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl removed owner|
|-|-|

Okay, I've decided to turn all the readmes into reST while I was at it, as this makes it look much nicer on GitHub and possibly PyPI if the release stuff is changed to add it as the long description.

I did a test mock release (./bin/admin/change-versions) to make sure that it updated all the rST READMEs under each project, and it did from what I can see.

It also adds stuff about the Sphinx docs.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @hawkowl|
|-|-|

Thanks.  I have some questions.

  1. Why do we have a `docs` and a `doc` directory?  That seems like a bad thing to me.
  1. Is `make html` really how people are supposed to build the docs?  Do we have automated test coverage for that?  We shouldn't tell people to do something unless we know it is going to work.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl removed owner|
|-|-|

Hi exarkun, thanks for the review.

1. is fixed, I think. It's something weird, there's only three pycs in there (why are pycs in there???)
2. I changed it to use ./bin/admin/build-docs, which there [is test coverage of https://buildbot.twistedmatrix.com/builders/documentation/builds/4405/steps/process-docs/logs/stdio]

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

> I changed it to use ./bin/admin/build-docs, which there is test coverage of ...

Indeed there is.  But `bin/admin` is for the release manager, not end users.  I wonder if we should be recommending that people use this in its current form.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban commented|
|-|-|

It would be nice to have instructions for building both local documentation and local API documentation so that I can do a few checks before sending changes for review.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_10"></a>|@glyph commented|
|-|-|

Replying to [exarkun](#note_8):
> > I changed it to use ./bin/admin/build-docs, which there is test coverage of ...
> 
> Indeed there is.  But `bin/admin` is for the release manager, not end users.  I wonder if we should be recommending that people use this in its current form.

Yup, we should.  If *only* the release manager uses it, then we won't have any forewarning when it breaks.  We're advising contributors to Twisted use it, not end users.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_11"></a>|@glyph set owner to @hawkowl|
|-|-|

Replying to [hawkowl](#note_5):
> Okay, I've decided to turn all the readmes into reST while I was at it, as this makes it look much nicer on GitHub and possibly PyPI if the release stuff is changed to add it as the long description.

I'm sorry to be obstructionist about this - I do appreciate the work you put in - but this is a massive and completely unrelated change.  Can you un-do this so someone can review the changes to the prose without looking at giant blobs of changes to release automation and formatting and multiple projects and such at the same time?  We can revisit switching README documentation to ReST later.  (And frankly I'd rather switch to Markdown if the motivation is to make it look nice on GH and such.)

Apropos of my previous reply to exarkun, maybe we should have a separate document for contributors and users; we do already have a `CONTRIBUTING`, after all.  If we're going to tell end users about docs, we should tell them about twisted.readthedocs.org or twistedmatrix.com/documents.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_12"></a>|@hawkowl commented|
|-|-|

(In [41979]) Branching to update-readme-6953-2.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_13"></a>|@hawkowl removed owner|
|-|-|

I've edited it to point people at http://twistedmatrix.com/documents/current/.

My reason for this is that I don't think people care about building the docs when they read Twisted's readme - they just want to read them.

I think we should have that documented better in the link given in CONTRIBUTING, for people that are going to be making changes to the docs. I'm not sure where, though - there's a lot of stuff in https://twistedmatrix.com/trac/wiki/ContributingToTwistedLabs .

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_14"></a>|@adiroiban commented|
|-|-|

I agree that people wanting to read Twisted documentation don't need to build the documentation.

-------

There was a talk about adding stuff to CONTRIBUTING on mailing list my feeling was that we should add more think to wiki and keep that file slim.

I have updated the wiki page https://twistedmatrix.com/trac/wiki/TwistedDevelopment#ToolsForDevelopment to include instructions about building docs and apidocs.

Maybe these 2 pages should be merged.

------

Instead or duplicating (incomplete) mailing list and IRC channel information, why not send people to https://twistedmatrix.com/trac/wiki/TwistedCommunity ?

Other than that, this looks good.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)|@khorn set owner to @khorn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_16"></a>|@khorn set owner to @hawkowl|
|-|-|

@hawkowl

Looks fine, feel free to merge this.  I think you're right about people not wanting to build the docs nearly so much as they want to read them.  Maybe we can put something about building the docs in once we decide definitively how we want to do that.

Also it doesn't make much sense to point users to local docs when they may not even be installed. (IIRC the Windows installer doesn't install docs)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_17"></a>|@hawkowl set status to `closed`|
|-|-|

(In [43084]) Merge update-readme-6953-2: Update the README for new docs

Author: hawkowl
Reviewers: exarkun, adiroiban, glyph, khorn
Fixes: [#6953](https://github.com/twisted/twisted/issues/6953)

Update the README to point to web-hosted docs, rather than local ones, and add a reference to #twisted.web.
